plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchangeModule.java

```diff
@@ -37,8 +38,9 @@ protected void setup(Binder binder)
     {
         binder.bind(FileSystemExchangeManager.class).in(Scopes.SINGLETON);
 
-        FileSystemExchangeConfig fileSystemExchangeConfig = buildConfigObject(FileSystemExchangeConfig.class);
-        String scheme = URI.create(requireNonNull(fileSystemExchangeConfig.getBaseDirectory(), "baseDirectory is null")).getScheme();
+        List<URI> baseDirectories = buildConfigObject(FileSystemExchangeConfig.class).getBaseDirectories();
+        verify(baseDirectories.stream().map(URI::getScheme).distinct().count() == 1, "Base directories should have the same scheme");
```

Comment:
`verify` is to check for assertions which are expected to never happen unless there is a bug in the code. Here we can legitly get multiple different schemas if user provides wrong configuration, right?
Please make it throw `TrinoException` with `CONFIGURATION_INVALID` error code and message which allows user (Trino cluster admin) to understand what is happenning.